include/deal.II/grid/tria_accessor.templates.h

```diff
@@ -2039,6 +2039,52 @@ TriaAccessor<structdim, dim, spacedim>::diameter () const
 }
 
 
+template <int structdim, int dim, int spacedim>
+std::pair<Point<spacedim>,double>
+TriaAccessor<structdim, dim, spacedim>::smallest_enclosing_sphere () const
+{
+  switch (structdim)
+    {
+    case 1:
+    {
+      const Point<spacedim> p10( this->vertex(1)-this->vertex(0));
+      return std::make_pair( this->vertex(0)+0.5*p10, p10.norm());
+    }
+    case 2:
+    {
+      const Point<spacedim> p30( this->vertex(3)-this->vertex(0));
+      const Point<spacedim> p21( this->vertex(2)-this->vertex(1));
+      return p30.norm() > p21.norm() ?
+             std::make_pair( this->vertex(0) + 0.5* p30, p30.norm()):
+             std::make_pair( this->vertex(1) + 0.5* p21, p21.norm());
```

Comment:
Taking p0(0|0), p1(.4,.4), p2(-.1, 1.1), p3(1|1), the diagonal p0p3 is larger than the diagonal p1p2
(1 vs. ~.99). Still, the distance between the middle of the diagonal ((.5|.5)) and p2 is larger than the distance between p0 and that midpoint (~.84 vs. 0.5)

Therefore, I doubt that this works as expected.